src/Suave/Logging/TraceHeader.fs

```diff
@@ -9,45 +9,38 @@ open Suave.Utils.RandomExtensions
 /// of possible values, so you can be fairly certain a given request
 /// id is unique, given a good random number generator.
 type TraceHeader =
-  { /// if this is the 'first' traced request, then trace_id equals
+  { /// If this is the 'first' traced request, then trace_id equals
     /// req_id. If it's the second, then trace_id = req_parent_id
     /// or otherwise third or later then trace_id, req_id and req_parent_id
     /// are all disjunct
-    trace_id      : uint64
-    /// the request id assigned when suave received the http request
+    traceId      : uint64
+
+    /// The request id assigned when suave received the http request
     /// In ZipKin/Dapper-speak, this is the span id
-    req_id        : uint64
+    reqId        : uint64
+
     /// possibly a parent
     /// In ZipKin/Dapper-speak, this is the span parent id
-    req_parent_id : uint64 option }
+    reqParentId : uint64 option }
+
+  static member traceId_     = Property (fun x -> x.traceId) (fun v x -> { x with traceId = v })
```

Comment:
I use (and wrote) Aether (https://github.com/xyncro/aether) for lensing in F#. The API for that is slightly non-F# ish and I'm actually considering a pending breaking change to fix that, but how it works isn't too likely to change now. It's based on the fairly naive implementation of lenses as composable get,set pairs, which is about the only option in F# short of language level changes (which, to be honest, would either have to be very feature-specific, which would make me hesitate, or very high level and general and "change to the type system" which doesn't seem too likely at this point). The naive implementation does work well enough though, and makes working with complex data structures more tractable/feasible.

Here's a (slightly nasty, but internal!) example from Hekate: https://github.com/xyncro/hekate/blob/master/src/Hekate/Hekate.fs#L177-L183 - note that the lenses are defined away from the types anyway, as they apply to type aliases. Here's a slightly nicer example where lenses are used to provide simple access to a more complex structure under the guise of some functions: https://github.com/kolektiv/freya/blob/develop/src/Freya.Machine/Configuration.fs - in that example one of the lenses being composed is defined as a static member on the type (as that's possible), but it doesn't feel like a convention that *quite* fits when other lenses won't be able to be. Also, from the consumer perspective, whether a lens is composed or not is opaque - it's just a lens, so distinguishing them, when all can be composed, etc. seems slightly awkward.

(As an aside, I'm about 95% sold on changing the functions such as getL, modL, setPL, etc. to things like Lens.get, Lens.map, and Lens.setPartial respectively. Increased verbosity and typing, but it is rather more the "F# way" - I think! I've gone more in that direction with more recent libs...)